fairlearn/postprocessing/_threshold_optimizer.py

```diff
@@ -228,6 +228,33 @@ def fit(self, X, y, *, sensitive_features, **kwargs):
             sensitive_feature_vector, y, scores)
         return self
 
+    def thresholds(self):
```

Comment:
I'm not convinced that this change belongs into this PR.

There are two main issues:
1. This should not be a method, but an [estimated attribute](https://scikit-learn.org/stable/developers/develop.html#estimated-attributes). Its name should match the name of the corresponding attribute of `InterpolatedThresholder`, which is `interpolation_dict` (that is not a great name and changing it to something like `thresholds` would be fine with me).
2. One of the use cases that came up with how folks have been using `ThresholdOptimizer` is that they need access to not just one set of thresholds, but to multiple--each corresponding to a different trade-off between the objective and the value of the constraint. For example, instead of simply minimizing error subject to false_positive_rate_parity, you may want to examine several solutions that achieve `false_positive_rate_parity` at different values of the `false_positive_rate`. This means that we will need a mechanism similar to the one we have right now in `GridSearch`. With that in mind, it might be confusing to have a field called `thresholds_` that captures just the "best set of thresholds" rather than the thresholds along the trade-off curve.

I think that (1) above would be very simple to address even in this PR. I'm not sure whether we need to address (2) right now or later, but I wanted to raise it for your awareness since it might inform how we rename the fields.

Reply:
(2) is definitely not in scope of this PR which is about documentation. This uncertainty about where this might eventually go if someone has time to get to it is what led me to make the original method start with an underscore to give us flexibility going forward. @adrinjalali 's comment about not using underscore-prefixed methods in the public docs made me change that.

(1) `InterpolatedThresholder` is internal. It doesn't even show up in the docs https://fairlearn.github.io/v0.5.0/api_reference/fairlearn.postprocessing.html
It's not a big deal to change the name, so I can change it there, too.


Given these points should it seems like it's best to keep it private AND still use it in the docs. @adrinjalali what do you think?

Reply:
Actually, if we use the name that takes form of a singular, then we could export it and be still future-proof re. (2) above. For example we could use something like `threshold_info_` or even `thresholder_info_`? We would similarly change the name of the corresponding field of `InterpolatedThresholder`.

Reply:
I'm usually wary of `attribute_` which is indeed a getter property instead of an actual attribute.

I would still find it odd to use private API in the docs. People copy/paste from the docs, and we should be stricter about changing what people use than what's internal. Putting something private there makes it used.

As an intermediary solution, I'd put the code for this function in the example and let people copy/paste the code while we figure the API. Or have a separate PR and figure the API out and then have this example merged.

Reply:
IMO `InterpolatedThresholder` is actually a great name and much better than `Thresholder`. I'd expect something very different from a class called `Thresholder`, namely a very simple thresholder. I can certainly just use the code in the example like @adrinjalali suggests. That would be
```
threshold_optimizer.interpolated_thresholder_.interpolation_dict
```
or perhaps `threshold_info_` instead of `interpolation_dict` as @MiroDudik suggests.

Reply:
Update: I've removed the method from `ThresholdOptimizer` and renamed `interpolation_dict` to `threshold_info_` on the `InterpolatedThresholder` class. In the example it just directly uses it now as mentioned above (except for the rename). This allows us to make this decision later on without requiring people to access private methods or attributes, and we can still show it. I hope that works for everyone.